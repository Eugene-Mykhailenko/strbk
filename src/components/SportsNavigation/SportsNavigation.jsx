import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Icon from '../Icon/Icon';
import styles from './SportsNavigation.scss';

const SportsNavigation = (props) => {
  const { activeItemId, items, onClick } = props;

  const getClassNames = (item) =>
    classNames(styles.item, {
      [styles.active]: item.id === activeItemId,
    });

  const renderItems = () =>
    items.map((item) => (
      <a
        href={item.link}
        className={getClassNames(item)}
        style={{
          borderColor: `var(--sportÐ¡olor_${activeItemId})`,
        }}
        key={item.id}
        onClick={(event) => onClick(event, item)}
      >
        <Icon width={32} height={24} glyph={item.icon} />
        <span className={styles.title}>{item.title}</span>
      </a>
    ));

  return <div className={styles.items}>{renderItems(props)}</div>;
};

SportsNavigation.styles = styles;

SportsNavigation.propTypes = {
  /** List of SportsNavigation items (id, title, icon) */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
      icon: PropTypes.string,
    }),
  ),
  /** Callback function for event */
  onClick: PropTypes.func.isRequired,
  /** Active item ID */
  activeItemId: PropTypes.string,
};

SportsNavigation.defaultProps = {
  items: [],
  activeItemId: undefined,
};

export default SportsNavigation;
