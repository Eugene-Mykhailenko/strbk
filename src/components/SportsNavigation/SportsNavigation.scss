@import '../../assets/styles/abstracts/variables';

.items {
  display: flex;
  overflow: hidden;
  overflow-x: auto;
}

.item {
  position: relative;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  flex: 1;
  min-width: 72px;
  padding: 7px 0;
  color: var(--color_6);
  cursor: pointer;
  transition: $transitionColor;

  &::after {
    content: '';
    position: absolute;
    bottom: 0;
    visibility: hidden;
    border-left: 6px solid transparent;
    border-right: 6px solid transparent;
    border-bottom: 6px solid;
    border-bottom-color: inherit;
    transform: translateY(6px);
    transition: $transitionTransform, $transitionBorderColor;
  }

  &.active {
    color: var(--color_3);

    &::after {
      transform: translateY(0);
      visibility: visible;
    }
  }

  .title {
    margin-top: 4px;
    font-size: 12px;
  }
}
