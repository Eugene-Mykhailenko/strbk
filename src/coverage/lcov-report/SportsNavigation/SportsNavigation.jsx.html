<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for SportsNavigation/SportsNavigation.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">SportsNavigation</a> SportsNavigation.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>12/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/12</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
&nbsp;
import Icon from '../Icon/Icon';
import styles from './SportsNavigation.scss';
&nbsp;
const SportsNavigation = (props) =&gt; {
  const { activeItemId, items, onClick } = props;
&nbsp;
  const getClassNames = (item) =&gt;
    classNames(styles.item, {
      [styles.active]: item.id === activeItemId,
    });
&nbsp;
  const renderItems = () =&gt;
    items.map((item) =&gt; (
      &lt;a
        href={item.link}
        className={getClassNames(item)}
        style={{
          borderColor: `var(--sport–°olor_${activeItemId})`,
        }}
        key={item.id}
        onClick={(event) =&gt; onClick(event, item)}
      &gt;
        &lt;Icon width={32} height={24} glyph={item.icon} /&gt;
        &lt;span className={styles.title}&gt;{item.title}&lt;/span&gt;
      &lt;/a&gt;
    ));
&nbsp;
  return &lt;div className={styles.items}&gt;{renderItems(props)}&lt;/div&gt;;
};
&nbsp;
SportsNavigation.styles = styles;
&nbsp;
SportsNavigation.propTypes = {
  /** List of SportsNavigation items (id, title, icon) */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
      icon: PropTypes.string,
    }),
  ),
  /** Callback function for event */
  onClick: PropTypes.func.isRequired,
  /** Active item ID */
  activeItemId: PropTypes.string,
};
&nbsp;
SportsNavigation.defaultProps = {
  items: [],
  activeItemId: undefined,
};
&nbsp;
export default SportsNavigation;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Sep 14 2018 16:36:00 GMT+0300 (EEST)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
