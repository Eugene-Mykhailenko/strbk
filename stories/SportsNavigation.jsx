import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withState } from '@dump247/storybook-state';

/* eslint-disable-next-line */
import SportsNavigation from '../src/components/SportsNavigation/SportsNavigation';
import storyStyles from './Stories.scss';

const addState = withState({ activeItemId: 'soccer' });
const addInfo = withInfo({
  text: `SportsNavigation component: useful in feed line.`,
  inline: true,
});
const props = {
  items: [
    {
      id: 'soccer',
      title: 'Soccer',
      icon: 'SI_Soccer',
    },
    {
      id: 'tennis',
      title: 'Tennis',
      icon: 'SI_Tennis',
    },
    {
      id: 'volleyball',
      title: 'Volleyball',
      icon: 'SI_BeachVolleyball',
    },
    {
      id: 'basketball',
      title: 'Basketball',
      icon: 'SI_Basketball',
    },
    {
      id: 'baseball',
      title: 'Baseball',
      icon: 'SI_Baseball',
    },
  ],
};

/* eslint-disable-next-line react/prop-types */
const wrapper = ({ store }) => (
  <div className={storyStyles.container}>
    <div className={storyStyles.col}>
      <SportsNavigation
        {...store.state}
        {...props}
        onClick={(event, item) => store.set({ activeItemId: item.id })}
      />
    </div>
  </div>
);

storiesOf('SportsNavigation', module).add(
  'SportsNavigation',
  addState(addInfo(wrapper)),
);
