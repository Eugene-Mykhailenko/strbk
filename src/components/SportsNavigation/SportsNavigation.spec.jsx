import React from 'react';
import SportsNavigation from './SportsNavigation';

const config = {
  items: [
    {
      id: 'soccer',
      title: 'Soccer',
      icon: 'SI_Soccer',
    },
    {
      id: 'basketball',
      title: 'Basketball',
      icon: 'SI_Basketball',
    },
    {
      id: 'tennis',
      title: 'Tennis',
      icon: 'SI_Tennis',
    },
    {
      id: 'hockey',
      title: 'Hockey',
      icon: 'SI_IceHockey',
    },
  ],
  onClick: jest.fn(),
};

const itemIds = ['soccer', 'basketball', 'tennis', 'hockey'];

describe('SportsNavigation', () => {
  it('should render items defined in the provided config', () => {
    const sportsNavigation = shallow(<SportsNavigation {...config} />);

    expect(sportsNavigation.children().length).toBe(4);
    expect(sportsNavigation).toMatchSnapshot();
  });

  it('should pass proper configuration for items', () => {
    const sportsNavigation = shallow(<SportsNavigation {...config} />);

    config.items.forEach((item, index) => {
      expect(sportsNavigation.props().children[index].key).toEqual(
        itemIds[index],
      );
    });

    expect(
      sportsNavigation
        .childAt(0)
        .childAt(1)
        .text(),
    ).toEqual(config.items[0].title);
    expect(
      sportsNavigation
        .childAt(0)
        .childAt(0)
        .props().glyph,
    ).toEqual(config.items[0].icon);
    expect(sportsNavigation).toMatchSnapshot();
  });

  it('should highlight active item', () => {
    const sportsNavigation = shallow(
      <SportsNavigation activeItemId={config.items[0].id} {...config} />,
    );

    expect(sportsNavigation.childAt(0).prop('className')).toMatch(/active/);
    expect(sportsNavigation).toMatchSnapshot();
  });

  it('should handle click event', () => {
    const sportsNavigation = mount(<SportsNavigation {...config} />);

    sportsNavigation
      .childAt(0)
      .childAt(0)
      .find('a')
      .simulate('click');

    expect(config.onClick).toHaveBeenCalledTimes(1);
    expect(config.onClick).toHaveBeenCalledWith(
      expect.any(Object),
      config.items[0],
    );

    config.onClick.mockClear();

    expect(sportsNavigation).toMatchSnapshot();
  });
});
