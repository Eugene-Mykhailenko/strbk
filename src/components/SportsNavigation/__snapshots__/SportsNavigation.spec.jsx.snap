// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SportsNavigation should handle click event 1`] = `
<SportsNavigation
  items={
    Array [
      Object {
        "icon": "SI_Soccer",
        "id": "soccer",
        "title": "Soccer",
      },
      Object {
        "icon": "SI_Basketball",
        "id": "basketball",
        "title": "Basketball",
      },
      Object {
        "icon": "SI_Tennis",
        "id": "tennis",
        "title": "Tennis",
      },
      Object {
        "icon": "SI_IceHockey",
        "id": "hockey",
        "title": "Hockey",
      },
    ]
  }
  onClick={[MockFunction]}
>
  <div
    className="items"
  >
    <a
      className="item"
      key="soccer"
      onClick={[Function]}
      style={
        Object {
          "borderColor": "var(--sportСolor_undefined)",
        }
      }
    >
      <Icon
        className=""
        fill="currentColor"
        glyph="SI_Soccer"
        height={24}
        width={32}
      >
        <svg
          className="icon"
          fill="currentColor"
          height={24}
          width={32}
        >
          <use
            xlinkHref="#SI_Soccer"
          />
        </svg>
      </Icon>
      <span
        className="title"
      >
        Soccer
      </span>
    </a>
    <a
      className="item"
      key="basketball"
      onClick={[Function]}
      style={
        Object {
          "borderColor": "var(--sportСolor_undefined)",
        }
      }
    >
      <Icon
        className=""
        fill="currentColor"
        glyph="SI_Basketball"
        height={24}
        width={32}
      >
        <svg
          className="icon"
          fill="currentColor"
          height={24}
          width={32}
        >
          <use
            xlinkHref="#SI_Basketball"
          />
        </svg>
      </Icon>
      <span
        className="title"
      >
        Basketball
      </span>
    </a>
    <a
      className="item"
      key="tennis"
      onClick={[Function]}
      style={
        Object {
          "borderColor": "var(--sportСolor_undefined)",
        }
      }
    >
      <Icon
        className=""
        fill="currentColor"
        glyph="SI_Tennis"
        height={24}
        width={32}
      >
        <svg
          className="icon"
          fill="currentColor"
          height={24}
          width={32}
        >
          <use
            xlinkHref="#SI_Tennis"
          />
        </svg>
      </Icon>
      <span
        className="title"
      >
        Tennis
      </span>
    </a>
    <a
      className="item"
      key="hockey"
      onClick={[Function]}
      style={
        Object {
          "borderColor": "var(--sportСolor_undefined)",
        }
      }
    >
      <Icon
        className=""
        fill="currentColor"
        glyph="SI_IceHockey"
        height={24}
        width={32}
      >
        <svg
          className="icon"
          fill="currentColor"
          height={24}
          width={32}
        >
          <use
            xlinkHref="#SI_IceHockey"
          />
        </svg>
      </Icon>
      <span
        className="title"
      >
        Hockey
      </span>
    </a>
  </div>
</SportsNavigation>
`;

exports[`SportsNavigation should highlight active item 1`] = `
<div
  className="items"
>
  <a
    className="item active"
    key="soccer"
    onClick={[Function]}
    style={
      Object {
        "borderColor": "var(--sportСolor_soccer)",
      }
    }
  >
    <Icon
      className=""
      fill="currentColor"
      glyph="SI_Soccer"
      height={24}
      width={32}
    />
    <span
      className="title"
    >
      Soccer
    </span>
  </a>
  <a
    className="item"
    key="basketball"
    onClick={[Function]}
    style={
      Object {
        "borderColor": "var(--sportСolor_soccer)",
      }
    }
  >
    <Icon
      className=""
      fill="currentColor"
      glyph="SI_Basketball"
      height={24}
      width={32}
    />
    <span
      className="title"
    >
      Basketball
    </span>
  </a>
  <a
    className="item"
    key="tennis"
    onClick={[Function]}
    style={
      Object {
        "borderColor": "var(--sportСolor_soccer)",
      }
    }
  >
    <Icon
      className=""
      fill="currentColor"
      glyph="SI_Tennis"
      height={24}
      width={32}
    />
    <span
      className="title"
    >
      Tennis
    </span>
  </a>
  <a
    className="item"
    key="hockey"
    onClick={[Function]}
    style={
      Object {
        "borderColor": "var(--sportСolor_soccer)",
      }
    }
  >
    <Icon
      className=""
      fill="currentColor"
      glyph="SI_IceHockey"
      height={24}
      width={32}
    />
    <span
      className="title"
    >
      Hockey
    </span>
  </a>
</div>
`;

exports[`SportsNavigation should pass proper configuration for items 1`] = `
<div
  className="items"
>
  <a
    className="item"
    key="soccer"
    onClick={[Function]}
    style={
      Object {
        "borderColor": "var(--sportСolor_undefined)",
      }
    }
  >
    <Icon
      className=""
      fill="currentColor"
      glyph="SI_Soccer"
      height={24}
      width={32}
    />
    <span
      className="title"
    >
      Soccer
    </span>
  </a>
  <a
    className="item"
    key="basketball"
    onClick={[Function]}
    style={
      Object {
        "borderColor": "var(--sportСolor_undefined)",
      }
    }
  >
    <Icon
      className=""
      fill="currentColor"
      glyph="SI_Basketball"
      height={24}
      width={32}
    />
    <span
      className="title"
    >
      Basketball
    </span>
  </a>
  <a
    className="item"
    key="tennis"
    onClick={[Function]}
    style={
      Object {
        "borderColor": "var(--sportСolor_undefined)",
      }
    }
  >
    <Icon
      className=""
      fill="currentColor"
      glyph="SI_Tennis"
      height={24}
      width={32}
    />
    <span
      className="title"
    >
      Tennis
    </span>
  </a>
  <a
    className="item"
    key="hockey"
    onClick={[Function]}
    style={
      Object {
        "borderColor": "var(--sportСolor_undefined)",
      }
    }
  >
    <Icon
      className=""
      fill="currentColor"
      glyph="SI_IceHockey"
      height={24}
      width={32}
    />
    <span
      className="title"
    >
      Hockey
    </span>
  </a>
</div>
`;

exports[`SportsNavigation should render items defined in the provided config 1`] = `
<div
  className="items"
>
  <a
    className="item"
    key="soccer"
    onClick={[Function]}
    style={
      Object {
        "borderColor": "var(--sportСolor_undefined)",
      }
    }
  >
    <Icon
      className=""
      fill="currentColor"
      glyph="SI_Soccer"
      height={24}
      width={32}
    />
    <span
      className="title"
    >
      Soccer
    </span>
  </a>
  <a
    className="item"
    key="basketball"
    onClick={[Function]}
    style={
      Object {
        "borderColor": "var(--sportСolor_undefined)",
      }
    }
  >
    <Icon
      className=""
      fill="currentColor"
      glyph="SI_Basketball"
      height={24}
      width={32}
    />
    <span
      className="title"
    >
      Basketball
    </span>
  </a>
  <a
    className="item"
    key="tennis"
    onClick={[Function]}
    style={
      Object {
        "borderColor": "var(--sportСolor_undefined)",
      }
    }
  >
    <Icon
      className=""
      fill="currentColor"
      glyph="SI_Tennis"
      height={24}
      width={32}
    />
    <span
      className="title"
    >
      Tennis
    </span>
  </a>
  <a
    className="item"
    key="hockey"
    onClick={[Function]}
    style={
      Object {
        "borderColor": "var(--sportСolor_undefined)",
      }
    }
  >
    <Icon
      className=""
      fill="currentColor"
      glyph="SI_IceHockey"
      height={24}
      width={32}
    />
    <span
      className="title"
    >
      Hockey
    </span>
  </a>
</div>
`;
